servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/TaskBasedAsyncSingleOperator.java

```diff
@@ -51,20 +52,33 @@ public String toString() {
         }
     };
 
+    private volatile boolean hasOffloaded;
```

Comment:
inline with comment above ... lets move this into the implementations to clarify expected thread access pattern and avoid shared state between subscribe/subscription when not necessary.

Reply:
I put it in the `TaskBasedAsync(Completable|Publisher|Single)Operator` because it is referenced by the `SubscribeOn::handleSubscribe` in addition to the signals offloading classes.

Reply:
https://github.com/apple/servicetalk/pull/1689#discussion_r674351132

> For example the SubscribeOn#handleSubscribe usage of offload doesn't need to share state with the Subscription (it will happen before, ordering doesn't really matter).

I'm not sure it needs to be shared. Does `SubscribeOn#handleSubscribe` have to care about the `hasOffloaded` state?